Add post on COVID reports and related code/packages (#342)
epiverse-trace/epiverse-trace.github.io: posts/covid-analysis/index.qmd

---
title: "How well prepared are we to rapidly analyse a new influenza pandemic? A brief perspective on analysis conducted for UK government advisory groups during COVID-19"
author:
  - name: "Adam Kucharski"
    orcid: "0000-0001-8814-9421"

> date: "2024-11-28"

Reviewer: ```suggestion
date: "2024-12-23"
```

Or any other date you would prefer, as right before holiday season can make things fall into a void 😊 Don't want your writing to get lost!

Author: Thanks! That date is fine – suspect we get relatively little direct traffic so can share on networks once everyone has recovered from festivities...

categories: [COVID, R, real-time analysis, pandemic preparedness, H5N1]
format:
  html:
    toc: true
---

> With [multiple reports](https://www.youtube.com/watch?v=i4frpQJMfn8) of influenza H5N1 cases that have no clear animal exposure, it is useful to consider what kinds of analysis would be required, and how easily this could be performed. As a starting point, this post reflections on some of the real-time work my colleagues and I contributed to inform the UK response to COVID-19.

Reviewer 2: ```suggestion
With [multiple reports](https://www.youtube.com/watch?v=i4frpQJMfn8) of influenza H5N1 cases that have no clear animal exposure, it is useful to consider what kinds of analysis would be required, and how easily this could be performed. As a starting point, this post reflects on some of the real-time work my colleagues and I contributed to inform the UK response to COVID-19.
```

These contributions typically fell into two main categories:

- Reports in response to specific questions from the Secretariat (e.g. exploring implications of policy options).
- Reports or preliminary results detailing broader epidemiological insights about COVID-19 my colleagues and I thought were noteworthy (e.g. unusual patterns with novel variants).

This post focuses on the analysis reports that I made a major contribution to as a member of SPI-M-O in the first 18 months after COVID was identified as a threat, i.e. between Jan 2020 and July 2021 (a full list has [previously been published](https://www.covid19.public-inquiry.uk/documents/inq000056464-rule-9-questionnaire-response-from-prof-adam-kucharski-scientific-advisory-group-for-emergencies-sage-scientific-pandemic-influenza-group-on-modelling-spi-m-o-undated/) by the COVID Inquiry). Narrative reports or analysis that did not involve substantial analytics or modelling (e.g. just direct plots of data) are not included here. I also include some major early piece of analysis that did not form SPI-M-O reports, but were published and informed subsequent analysis and modelling.

> If another pandemic were to hit, how easily and quickly could we do this analysis again? To document where we currently are, it is important to understand where potential gaps and bottlenecks are. For each report, I therefore review three main criteria:

Reviewer 2: ```suggestion
If another pandemic were to hit, how easily and quickly could we do these analyses again? To document where we currently are, it is important to understand where potential gaps and bottlenecks are. For each report, I therefore review three main criteria:
```

1. **Code availability**: is the original analysis code public? (With link if relevant, or context if not.)
2. **Package availability**: is the analysis code or underlying method currently packaged up for easy reuse?

> 3. **Off-the-shelf readiness in R**: can we put in future transmissible H5N1 characteristics and run almost straight away?

Author: Will address below in summary response

The overall effectiveness of the UK response, and which areas could be strengthened, are topics currently being examined by the COVID Inquiry, and will not be covered here. If readers are interested, there are some broader reflections from members of the UK modelling community, and recommendations for improvements, in [Sherratt et al, 2024, Wellcome Open Research](https://wellcomeopenresearch.org/articles/9-12/v1?s3BucketUrl=https%3A%2F%2Fwellcomeopenresearch.s3.eu-west-1.amazonaws.com&gtmKey=GTM-5P673KJ&submissionUrl=%2Ffor-authors%2Fpublish-your-research&otid=23226e40-fdd0-4acd-97a3-d9bad93befed&immUserUrl=https%3A%2F%2Fwor-proxy.f1krdev.com%2Feditor%2Fmember%2Fshow%2F).

### 28 Jan 2020: Early estimation of transmissibility and control

[Code](https://github.com/adamkucharski/2020-ncov) ✅  | Package ⏳ | Off-shelf ❌

> This analysis in early 2020 focused on estimation of transmissiblity and subsequent effect of lockdown control measures, and brought together reported cases in China, exported cases identified internationally, and infections detected on evacuation flights. A stochastic SEIR model was fitted with SMC to estimate how $R_t$ varied over time, to distinguish between an epidemic that ended because of control vs immunity. Later published as [Kucharski et al, Lancet Inf Dis, 2020](https://www.thelancet.com/article/S1473-3099(20)30144-4/fulltext).

Author: Done

### 2 Feb 2020: Early analysis of contact tracing effectiveness

[Code](https://github.com/cmmid/ringbp) ✅  | [Package](https://github.com/epiforecasts/ringbp) ✅  | Off-shelf ⏳

Another early analysis was a paper for SPI-M-O: **Feasibility of controlling 2019-nCoV outbreaks by isolation of cases and contacts**, later published as [Hellewell et al, Lancet Global Health](https://www.thelancet.com/article/S2214-109X(20)30074-7/fulltext). It used a branching process model to explore how transmission (e.g. $R_0$ and % presymptomatic transmission) and control parameters (e.g. % contacts traced) could influence the risk of a large outbreak.

> **Suggested or ongoing development:** There are several issues on the [\{ringbp\}](https://github.com/epiforecasts/ringbp) package repo that, once complete, would allow for faster implementation for new pathogens, especially in combination with [\{epiparameter\}](https://epiverse-trace.github.io/epiparameter/). The [\{epichains\}](https://epiverse-trace.github.io/epichains/) package also allows for estimation of simpler branching processes (i.e. without targeted control like contact tracing).

Author: Thanks, will add

### 3 Mar 2020: Early estimation of severity

[Code](https://github.com/thimotei/cCFRDiamondPrincess) ✅  | [Package](https://github.com/epiverse-trace/cfr/) ✅  | Off-shelf ⏳

This analysis estimated the infection and case fatality ratio by age for COVID-19 using age-adjusted data from the outbreak on the Diamond Princess cruise ship. Later published as [Russell et al, Eurosurveillance, 2020](https://www.eurosurveillance.org/content/10.2807/1560-7917.ES.2020.25.12.2000256).

The methods used for this analysis are now implemented as the Epiverse-TRACE [\{cfr\}](https://github.com/epiverse-trace/cfr/) package. There is also a ['how to' example for age-specific CFR estimation](https://epiverse-trace.github.io/howto/analyses/estimate_severity/cfr-stratified-severity.html).

**Suggested or ongoing development:** There are several [issues](https://github.com/epiverse-trace/cfr/issues) open that aim to strengthen \{cfr\}, especially for edge cases (e.g. very high CFR) or uncertainty when estimating underascertainment.

### 11 Mar 2020: pre-COVID social mixing patterns

> [Data](https://www.medrxiv.org/content/10.1101/2020.02.16.20023754v2.supplementary-material) ✅  | Package ❌ |  Off-shelf ❌

Author: Good point – will address below in summary response

### 3 Mar 2020 onwards: Early population-level scenarios

[Code](https://github.com/cmmid/covid-uk) ✅  | Package ✅  | Off-shelf ⏳

A collection of population-level scenario modelling reports generated between February and April 2020 was later published as a summary paper, [Davies et al, Lancet Public Health, 2020](https://www.thelancet.com/journals/lanpub/article/PIIS2468-2667(20)30133-X/fulltext). The original model, known as `coivdm`, had a code base that would be reused for multiple epidemic waves, including novel variants. However, many the basic scenarios can be now be explored using the Epiverse-TRACE [\{epidemics\}](https://epiverse-trace.github.io/epidemics/). In particular, \{epidemics\} can simulate scenarios with multiple overlapping interventions targeting different age groups, and reflect uncertainty in $R_0$. A version of this package has already been used to [project future outbreak scenarios in Gaza](https://gaza-projections.org/).

Some further reflections on specific pieces of COVID analysis for SPI-M-O below:

> - 3 March. [**Effect of social distancing measures on deaths & peak demand for hospitals in England.**](https://www.gov.uk/government/publications/the-effect-of-social-distance-measures-on-deaths-and-peak-demand-for-hospital-services-in-england-3-march-2020) **Suggested or ongoing development:** Although \{epidemics\} is set up to simulate infections, there is currently no database for burden to draw on (e.g. easy conversion to hospitalisations and deaths). The WHO Hub [simex model](https://github.com/WHO-Collaboratory/simex) has some functions for COVID, but this could be a useful feature. Burden estimation is also planned in the [Epiverse-TRACE late stage tutorials](https://github.com/epiverse-trace/tutorials-late/issues/54).

Author: Have edited – meant to refer to burden conversion

- 11 March. [**Impact of agressively managing peak incidence.**](https://www.gov.uk/government/publications/the-impact-of-aggressively-managing-peak-incidence-11-march-2020
). **Suggested or ongoing development:** This analysis included epidemic-dependent triggers (e.g. close venues once hospitalisations hit a certain level), which are not yet included in \{epidemics\}.
- 11 March. [**Impact of banning sporting events and other leisure activities on Covid-19 epidemic.**](https://www.gov.uk/government/publications/the-impact-of-banning-sporting-events-and-other-leisure-activities-on-the-covid-19-epidemic-11-march-2020)
- 17 March. [**Impact of adding school closure to other social distance measures.**](https://www.gov.uk/government/publications/the-impact-of-adding-school-closure-to-other-social-distance-measures-17-march-2020)
- 1 April. [**Impact of relaxing lockdown measures.**](https://www.gov.uk/government/publications/the-impact-of-relaxing-lockdown-measures-2-1-april-2020)
- 3 April. [**Modelling scenarios for relaxation of lockdown in England.**](https://www.gov.uk/government/publications/lshtm-modelling-scenarios-for-relaxation-of-lockdown-in-england-3-may-2020)

> **Suggested or ongoing development:** There is work in progress with \{epidemics\} to allow a more flexible and editable [\{odin\} back end](https://github.com/epiverse-trace/epidemics/issues/254), in case different features are required in future.

Author: Good idea - have added
